subsys/net/lib/lwm2m/lwm2m_obj_firmware.c

```diff
@@ -116,9 +116,10 @@ void lwm2m_firmware_set_update_state(uint8_t state)
 			update_state, state);
 	}
 
-	update_state = state;
+	lwm2m_engine_set_u8("5/0/3", state);
+
 	NOTIFY_OBSERVER(LWM2M_OBJECT_FIRMWARE_ID, 0, FIRMWARE_STATE_ID);
```

Comment:
nit: `NOIFY_OBSERVER()` is redundant if `lwm2m_eninge_set_u8()` - the engine will take care of this as well.